functions/src/services/GoogleSheets.ts

```diff
@@ -0,0 +1,199 @@
+      range: this.sheetName + this._computeRange(limit),
+    });
+
+    const { statusText, status, data } = targetSheet;
+    if (statusText !== 'OK' || status !== 200) {
+      console.error('Error: Not able to get google sheet');
+      return null;
+    }
+    const { majorDimension, values } = data;
+    if (majorDimension !== IMajorDimensions.Rows || !values || !values.length) {
+      console.error('Error: Values are wrong format');
+      return null;
+    }
+
+    this.headers = values.shift();
+    return this._convertRows(values);
+  }
+
+  protected async getHeaders() {
+    if (this.headers && this.headers.length) {
+      return;
+    }
+    const firstRow: any = await this.connection.spreadsheets.values.get({
+      spreadsheetId: this.spreadsheetId,
+      majorDimension: IMajorDimensions.Rows,
+      range: `${this.sheetName}!1:1`,
+    });
+    const { statusText, status, data } = firstRow;
+    if (statusText !== 'OK' || status !== 200) {
+      console.error('Error: Not able to get google sheet');
+      return null;
+    }
+    this.headers = data.values[0];
+  }
+
+  public async getColumn(columnName: string) {
+    await this.connect();
+    await this.getHeaders();
+
+    let targetedColumn;
+    const index = this.headers.indexOf(columnName);
+    if (index > -1) {
+      targetedColumn = this._getNotationLetterFromIndex(index);
+    } else {
+      throw Error('Column not found');
+    }
+
+    const entireColumn: any = await this.connection.spreadsheets.values.get({
+      spreadsheetId: this.spreadsheetId,
+      majorDimension: IMajorDimensions.Rows,
+      range: `${this.sheetName}!${targetedColumn}:${targetedColumn}`,
+    });
+    return [].concat.apply([], entireColumn.data.values);
+  }
+
+  public async getRow(rowNumber: number): Promise<any> {
+    this.connect();
+    this.getHeaders();
+
+    const rowRange = `${this.sheetName}!${rowNumber}:${rowNumber}`;
+    const row: any = await this.connection.spreadsheets.values.get({
+      spreadsheetId: this.spreadsheetId,
+      majorDimension: IMajorDimensions.Rows,
+      range: rowRange,
+    });
+    return this._convertRows(row.data.values)[0];
+  }
+
+  public async updateRow(rowNumber: number, updateValues: any): Promise<any> {
+    this.connect();
+    this.getHeaders();
+
+    const rowRange = `${this.sheetName}!${rowNumber}:${rowNumber}`;
+    const updateRow = this._convertColumnFormat(updateValues);
+    const afterUpdate = await this.connection.spreadsheets.values.update({
+      spreadsheetId: this.spreadsheetId,
+      range: rowRange,
+      valueInputOption: IValueInputOption.USER_ENTERED,
+      resource: {
+        values: [updateRow],
+      },
+    });
+    return afterUpdate;
+  }
+
+  /**
+   * Add a new row to specified spread sheet
+   *
+   * @param sheetId The sheet to query from google sheets api
+   * @param appendValues Data values to add to Google Sheets
+   */
+  public async appendRow<T>(appendValues: T): Promise<any> {
```

Comment:
What is the use of generic here, and why it was not used in `updateRow`?

Reply:
The generic is for if the end user wants to pass in a interface for strong typing, VSCode completion. For example you could: `gsheets.appendRow<IAllocationRows>(appendValues)`

We could use a generic for `updateRow` as well. Currently it's just set to `any`.